packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -0,0 +1,269 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of material_animated_icons;
+
+// The code for drawing animated icons is kept in a private API, as we are not
+// yet ready for exposing a public API for (partial) vector graphics support.
+// See: https://github.com/flutter/flutter/issues/1831 for details regarding
+// generic vector graphics support in Flutter.
+
+/// Shows an animated icon at a given animation [progress].
+///
+/// The available icons are specified in [AnimatedIcons].
+class AnimatedIcon extends StatelessWidget {
+
+  /// Creates an AnimatedIcon.
+  ///
+  /// [progress], and [icon] cannot be null.
+  /// The [size] and [color] default to the value given by the current [IconTheme].
+  const AnimatedIcon({
+    Key key,
+    @required this.progress,
+    @required this.icon,
+    this.color,
+    this.size,
+    this.semanticLabel,
+    this.textDirection,
+    // TODO(amirh): add a parameter for controlling scaling behavior.
+  }) : assert(progress != null),
+       assert(icon != null);
+
+  /// The animation progress for the animated icon.
+  /// The value is clamped to be between 0 and 1.
+  ///
+  /// This determines the actual frame that is displayed.
+  final Animation<double> progress;
+
+  /// The color to use when drawing the icon.
+  ///
+  /// Defaults to the current [IconTheme] color, if any.
+  ///
+  /// The given color will be adjusted by the opacity of the current
+  /// [IconTheme], if any.
+  ///
+  /// If no [IconTheme]s are specified, icons will default to black.
+  ///
+  /// In material apps, if there is a [Theme] without any [IconTheme]s
+  /// specified, icon colors default to white if the theme is dark
+  /// and black if the theme is light.
```

Comment:
This paragraph contradicts the previous paragraph.

Reply:
copy-paste from icons.dart :) updated in both files